tests/fov_watcher_test.py

```diff
@@ -127,6 +127,103 @@ def test_run_structure(run_json, expected_files, recwarn):
+        json_object=COMBINED_RUN_JSON_SPOOF,
+    )
+    for tissue_file in sorted(
+        [f for f in os.listdir(COMBINED_DATA_PATH) if ".bin" in f or ".json" in f]
+    ):
+        shutil.copy(
+            os.path.join(COMBINED_DATA_PATH, tissue_file),
+            os.path.join(run_folder_path, tissue_file),
+        )
+
+
+@patch("toffy.watcher_callbacks.visualize_qc_metrics", side_effect=mock_visualize_qc_metrics)
+@patch("toffy.watcher_callbacks.visualize_mph", side_effect=mock_visualize_mph)
+@pytest.mark.parametrize("run_folder_lag", [0, 5, 15])
+@parametrize_with_cases(
+    "run_cbs,int_cbs,fov_cbs,kwargs,validators,watcher_start_lag,existing_data",
+    cases=WatcherCases.case_default,
+)
+def test_watcher_run_timeout(
+    mock_viz_qc,
+    mock_viz_mph,
+    run_cbs,
+    int_cbs,
+    fov_cbs,
+    kwargs,
+    validators,
+    watcher_start_lag,
+    existing_data,
+    run_folder_lag,
+):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tiff_out_dir = os.path.join(tmpdir, "cb_0", RUN_DIR_NAME)
+        qc_out_dir = os.path.join(tmpdir, "cb_1", RUN_DIR_NAME)
+        mph_out_dir = os.path.join(tmpdir, "cb_2", RUN_DIR_NAME)
+        plot_dir = os.path.join(tmpdir, "cb_2_plots", RUN_DIR_NAME)
+        pulse_out_dir = os.path.join(tmpdir, "cb_3", RUN_DIR_NAME)
+        stitched_dir = os.path.join(tmpdir, "cb_0", RUN_DIR_NAME, f"{RUN_DIR_NAME}_stitched")
+
+        # add directories to kwargs
+        kwargs["tiff_out_dir"] = tiff_out_dir
+        kwargs["qc_out_dir"] = qc_out_dir
+        kwargs["mph_out_dir"] = mph_out_dir
+        kwargs["pulse_out_dir"] = pulse_out_dir
+        kwargs["plot_dir"] = plot_dir
+
+        # ensure warn_overwrite set to False if intermediate callbacks set, otherwise True
+        kwargs["warn_overwrite"] = True if int_cbs else False
+
+        run_folder = os.path.join(tmpdir, "test_run")
+        log_out = os.path.join(tmpdir, "log_output")
+        fov_callback, run_callback, intermediate_callback = build_callbacks(
+            run_cbs, int_cbs, fov_cbs, **kwargs
+        )
+
+        with Pool(processes=4) as pool:
+            pool.apply_async(_slow_create_run_folder, (run_folder, run_folder_lag))
+
+            if run_folder_lag > 10:
+                with pytest.raises(FileNotFoundError, match=f"Timed out waiting for {run_folder}"):
+                    res_scan = pool.apply_async(
+                        start_watcher,
+                        (
+                            run_folder,
+                            log_out,
+                            fov_callback,
+                            run_callback,
+                            intermediate_callback,
+                            10,
+                            1,
+                            SLOW_COPY_INTERVAL_S,
+                        ),
+                    )
+
+                    res_scan.get()
+            else:
+                res_scan = pool.apply_async(
```

Comment:
Just for test timing, I'm wondering if we need to see this watcher test to completion? Or is it sufficient to just exit out of start_watcher once the run folder has been created successfully and the script has successfully began to create outputs from the first few callbacks? 
If the watcher tests don't take that long then this is probably fine as is.

Reply:
Looks like the workflow isn't starting the test process, but I am able to manually run it with.